### DISCUSSION TITLE
How can I add more categories to the primaryCategory list in Channel Read?

### DISCUSSION THREAD

**Discussion by rajnishdargan** [Sunbird-Knowlg/Community#224] [Q&A] [answered]
**Context:**
I created a channel using the Channel V3 create API and also added a few Object Categories and Category Definitions.

When I use the channel read API, I can see some primary categories  showing up, like:
```
primaryCategories: [
    {
        "identifier": "obj-cat:certificate-template_asset_all",
        "name": "Certificate Template",
        "targetObjectType": "Asset"
    },
    {
        "identifier": "obj-cat:course_collection_all",
        "name": "Course",
        "targetObjectType": "Collection"
    },
    {
        "identifier": "obj-cat:practice-question-set_questionset_all",
        "name": "Practice Question Set",
        "targetObjectType": "QuestionSet"
    },
    {
        "identifier": "obj-cat:test-category_content_all",
        "name": "Test Category",
        "targetObjectType": "Content"
    }
]
```
However, I recently created new Object Categories and Definitions for "Learning Resource" and "eTextbook", but they are not showing up in the primaryCategory list when I perform a channel read. Below are the details of the object category and definition I created:

**Object Category:**

```
{
    "request": {
        "objectCategory": {
            "name": "Learning Resource",
            "description": "Learning Resource"
        }
    }
}
```

**Object Category Definition:**
```
{
    "request": {
        "objectCategoryDefinition": {
            "categoryId": "obj-cat:learning-resource",
            "targetObjectType": "Content",
            "objectMetadata": {
                "config": {},
                "schema": {}
            },
            "forms": {}
        }
    }
}
```

**Issue:**
The "Learning Resource" category is not appearing in the primaryCategory list when I read the channel. I want to bring this category (and others like "eTextbook") into the primaryCategory list during channel read.

**Question:**
What are the steps to update or add more categories like "Learning Resource" and "eTextbook" to the primaryCategory list in a channel?

**Here is complete channel Read Response:**
[channelRead.json](https://github.com/user-attachments/files/17473139/channelRead.json)

@pallakartheekreddy @HarishGangula 
@coolbung @itsvick @snehal0904 @Aar-if

**Comment by pallakartheekreddy** [comment]
@rajnishdargan As discussed, Please update the discussion thread.

**Reply by rajnishdargan** [reply, marked as answer]
@pallakartheekreddy

This is working now, it seems there was a sync issue, similar to the search indexer issue. Thanks for guiding me on updating the primaryCategory list in the channel and for connecting over the call to discuss this.
